core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -202,7 +209,7 @@ object SparkSubmit extends CommandLineUtils {
           printWarning(s"Master endpoint ${args.master} was not a REST server. " +
             "Falling back to legacy submission gateway instead.")
           args.useRest = false
-          submit(args)
+          submit(args, false)
```

Comment:
This makes an assumption that the class in the original submit call initialized logging; it should be ok, since this case is only hit in certain cases when submitting to standalone / mesos in cluster mode, and if you miss a few log messages there it's not a big deal.